File: 2024 December/Number of occurrence/main.cpp
// Link: https://www.geeksforgeeks.org/problems/number-of-occurrence2259/1

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int countFreq(vector<int> &arr, int target)
    {
        int n = arr.size();
        int c = 0;
        for (int i = 0; i < n; i++)
        {
            if (arr[i] == target)
                c++;
        }
        return c;
    }
};

int main()
{
    Solution object;
    vector<int> arr = {1, 1, 2, 2, 2, 2, 3};
    int target = 2;
    cout << object.countFreq(arr, target) << endl;

    arr = {1, 1, 2, 2, 2, 2, 3};
    target = 4;
    cout << object.countFreq(arr, target) << endl;

    arr = {8, 9, 10, 12, 12, 12};
    target = 12;
    cout << object.countFreq(arr, target) << endl;
    return 0;
}

File: 2024 December/Number of occurrence/main.py
# Link: https://www.geeksforgeeks.org/problems/number-of-occurrence2259/1


class Solution:
    def countFreq(self, arr, target):
        c = 0
        for num in arr:
            if num == target:
                c += 1
        return c


if __name__ == "__main__":
    object = Solution()
    arr = [1, 1, 2, 2, 2, 2, 3]
    target = 2
    print(object.countFreq(arr, target))

    arr = [1, 1, 2, 2, 2, 2, 3]
    target = 4
    print(object.countFreq(arr, target))

    arr = [8, 9, 10, 12, 12, 12]
    target = 12
    print(object.countFreq(arr, target))
